llvm/lib/Target/AMDGPU/AMDGPUTargetMachine.cpp

```diff
@@ -199,11 +199,6 @@ static VGPRRegisterRegAlloc fastRegAllocVGPR(
   "fast", "fast register allocator", createFastVGPRRegisterAllocator);
 } // anonymous namespace
 
-static cl::opt<bool>
-EnableEarlyIfConversion("amdgpu-early-ifcvt", cl::Hidden,
-                        cl::desc("Run early if-conversion"),
-                        cl::init(false));
```

Comment:
I'd expect most if conversion to be done on the IR level. What cases do we need more aggressive MIR conversion for?

Can this be done in 2 steps? One showing the benefit of turning this one, and then another with the replacement pass?

Reply:
I have one main concern about doing this at IR level: How to handle instructions accessing memory.
I guess it could be done through masked load/store operations. But what if we have some other built-in?

I'm also not sure about how to implement the cost model in the LLVM-IR, but I guess it could be possible.

Reply:
PS: For the moment this pass hasn't covered the cases the early-ifcvt pass covers. I should add it back.